app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/claims/CheckMovementReferenceNumbersController.scala

```diff
@@ -63,27 +75,27 @@ class CheckMovementReferenceNumbersController @Inject() (
 
   def deleteMrn(mrnIndex: AssociatedMrnIndex): Action[AnyContent] = authenticatedActionWithSessionData.async {
     implicit request =>
-      withAnswers[AssociatedMRNsAnswer] { (fillingOutClaim, maybeAssociatedMRNs) =>
-        def redirectToShowMrnsPage() =
-          Redirect(routes.CheckMovementReferenceNumbersController.showMrns())
-
-        val idx = mrnIndex.toRegular
-
-        val maybeUpdatedClaim = for {
-          mrns    <- maybeAssociatedMRNs.map(_.toList)
-          updated <- AssociatedMRNsAnswer(mrns.take(idx) ++ mrns.drop(idx + 1))
-          claim    = FillingOutClaim.of(fillingOutClaim)(_.copy(associatedMRNsAnswer = updated.some))
-        } yield claim
+      withAnswers[AssociatedMRNsAnswer] { (fillingOutClaim, associatedMRNsAnswer) =>
+        withAnswers[AssociatedMRNsDeclarationAnswer] { (_, associatedMRNsDeclarationAnswer) =>
```

Comment:
How about just extracting FillingOutClaim? Like I did for `showMrns`
```
request.using({ case journey: FillingOutClaim =>
```
Then you can do something like
```
journey.draftClaim.MRNs.combineWithDeclarations
journey.draftClaim.associatedMRNsAnswer
journey.draftClaim.associatedMRNsDeclarationAnswer
```
No need for `dataExtractor1`, `dataExtractor2`

Reply:
make sense